src/Prometheus.Client.MetricServer/BaseMetricServer.cs

```diff
@@ -20,19 +21,21 @@ public abstract class BaseMetricServer
         /// <param name="registry">Collector registry</param>
         /// <param name="collectors">IOnDemandCollectors</param>
         /// <param name="useDefaultCollectors">Use default collectors</param>
-        protected BaseMetricServer(ICollectorRegistry registry, List<IOnDemandCollector> collectors, bool useDefaultCollectors)
+        protected BaseMetricServer(ICollectorRegistry registry, List<ICollector> collectors, bool useDefaultCollectors)
```

Comment:
Collectors parameter could be removed, as client code can modify ICollectorRegistry directly (it was not possible in v2, so we had to have extra parameter)

Reply:
fixed!